src/cupbearer/detectors/activation_based.py

```diff
@@ -0,0 +1,285 @@
+from collections import defaultdict
+from pathlib import Path
+from typing import Any, Callable
+
+import torch
+import tqdm
+from torch.utils.data import Dataset
+
+from cupbearer import utils
+from cupbearer.data import MixedData
+
+from .anomaly_detector import AnomalyDetector
+
+
+def model_checksum(model: torch.nn.Module) -> float:
+    """Hacky but fast way to get a checksum of the model parameters.
+    Just sums the absolute values.
+    """
+    device = next(model.parameters()).device
+    # Using float64 to get some more bits and make collisions less likely.
+    param_sum = torch.tensor(0.0, dtype=torch.float64, device=device)
+    for param in model.parameters():
+        param_sum += param.abs().sum()
+    return param_sum.item()
```

Comment:
But the parameters are only 32 bit, right? You remove risk of overflows, but shouldn't collision be just as likely?

Also, I'm slightly concerned about floating point errors here, in case you cache the activations on one machine and then run on a different one with different instruction sets.

You could try:
```suggestion
    # Using int64 to get some more bits and make collisions less likely.
    param_sum = torch.tensor(0, dtype=torch.int64, device=device)
    for param in model.parameters():
        param_sum += param.view(torch.int32).to(torch.int64).sum()
    return param_sum.item()
```

Reply:
I've just removed the checksum entirely. There's a similar problem with `activation_preprocessing_func` anyway. I guess we could look at the code for the function and use that, but that won't easily work for lambdas, overall it just seems way too complex. (And I don't just want to use the `id` because that won't work for loading a cache from disk.) So better to warn users that it's their responsibility